src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocationsTests.cs

```diff
@@ -2002,63 +2000,94 @@ async void M(CancellationToken ct)
             return VerifyCS.VerifyCodeFixAsync(originalCode, fixedCode);
         }
 
-        [Theory]
-        [InlineData("CancellationToken c", "", "", "")]
-        [InlineData("", "CancellationToken c", "", "")]
-        [InlineData("", "", "<CancellationToken>", "c")]
-        public Task CS_Diagnostic_MultiNesting(string topMethodParam, string localMethodParam, string extensionTypeParam, string extensionArg)
+        [Fact]
+        public Task CS_Diagnostic_MultiNesting_TopMethod()
         {
             string originalCode = $@"
 using System;
 using System.Threading;
 using System.Threading.Tasks;
-public static class Extensions
+class C
 {{
-    public static void Extension(this bool b, Action{extensionTypeParam} action) {{}}
+    private readonly object lockingObject = new object();
+    public void TopMethod(CancellationToken c)
+    {{
+        void LocalMethod()
+        {{
+            bool b = false;
+            lock (lockingObject)
+            {{
+                [|TokenMethod|]();
+            }}
+        }}
+    }}
+    void TokenMethod(CancellationToken ct = default) {{}}
 }}
+            ";
+            string fixedCode = $@"
+using System;
+using System.Threading;
+using System.Threading.Tasks;
 class C
 {{
     private readonly object lockingObject = new object();
-    public void TopMethod({topMethodParam})
+    public void TopMethod(CancellationToken c)
     {{
-        void LocalMethod({localMethodParam})
+        void LocalMethod()
         {{
             bool b = false;
-            b.Extension(({extensionArg}) =>
+            lock (lockingObject)
             {{
-                lock (lockingObject)
-                {{
-                    [|TokenMethod|]();
-                }}
-            }});
+                TokenMethod(c);
```

Comment:
Do we have a test where this is a static local function? I presume your code fix should bail out for that, and we only get a diagnostic?

Reply:
That's a very good edge case.
Let me check. If I don't find such test, I'll add it.
I agree, it should bail out. If it doesn't, I'll fix it.

Reply:
I didn't have a test for that, so I added it. The analyzer wasn't able to handle this case, so I fixed it.